<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Data &amp; Scripts â€“ TransIsthmian Shrimp</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-listing/list.min.js"></script>
<script src="../../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-med-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: ['listing-name','listing-description',{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-name-sort'] }],
      
      searchColumns: ["listing-name","listing-description"],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-med-listing'] = new List('listing-med-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../assets/icon.png" alt="" class="navbar-logo"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item compact">
    <a class="nav-link" href="../../index.html"> <i class="bi bi-home" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../docs/data/index.html" aria-current="page"> 
<span class="menu-text">Data &amp; Scripts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../docs/ssu-workflows/index.html"> 
<span class="menu-text">16S rRNA</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../docs/mg-workflows/index.html"> 
<span class="menu-text">Metagenomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../docs/pub/index.html"> 
<span class="menu-text">Paper</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/istmobiome/trans-shrimp/"> <i class="bi bi-github" role="img" aria-label="TransI-Shrimp">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../about.html"> <i class="bi bi-lightbulb-fill" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Data &amp; Scripts</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/data/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data &amp; Scripts</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../docs/ssu-workflows/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16S rRNA Processing</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/ssu-workflows/sampledata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Sample Metadata</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/ssu-workflows/asv/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. ASV Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/ssu-workflows/otu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. OTU Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/ssu-workflows/med/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. MED Workflow</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../docs/mg-workflows/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metagenomics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/mg-workflows/setup/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Annotation Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/mg-workflows/assembly/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Co-Assembly &amp; Annotations</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../docs/pub/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publication Material</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/pub/references/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflow References</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/pub/submitdata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Submitting Data</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content page-columns page-full column-page-right" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">Data &amp; Scripts</h1>
</div>



<div class="quarto-title-meta column-page-right">

    
  
    
  </div>
  


</header><div class="column-body-outset-right">
<p>Quick access to pipeline processing scripts and raw data. With these scripts and associated data you can run the processing steps for each analysis.</p>
<p>All sequence data is linked to projects on the <a href="https://www.ebi.ac.uk/ena/browser/">European Nucleotide Archive (ENA)</a>. The scripts and associated files can be downloaded as <code>.zip</code> archives by clicking the links. Or if you prefer, you can copy the scripts and files from the code blocks below.</p>
</div>
<div class="column-page-inset-right">

<style type="text/css">
.nav-tabs {
  margin-top: 0.5rem;
  border-bottom: none;
}

.callout {
  margin-top: 0;
}

.nav-tabs .nav-link {
  text-align: center;
  margin-right: 15px;
  margin-top: 10px;
  width: 147px;
  font-size: 0.9em;
  font-weight: 600;
}

.nav-tabs .nav-link, 
.nav-tabs .nav-link.active, 
.nav-tabs .nav-item.show .nav-link {
  border: 1px solid  rgb(222, 226, 230);
  border-radius: 10px;
  color: rgb(80,146,221);
}
.nav-tabs .nav-link:hover {
   border-color: rgb(80,146,221);
   border-width: 1px;
} 

.nav-tabs .nav-link.active, 
.nav-tabs .nav-item.show .nav-link {
  border-color: rgb(80,146,221);
  border-width: 2px;
}

.nav-tabs .nav-link i {
  display: block;
  font-size: 3rem;
  color: rgb(80,146,221);
  margin-bottom: 5px;
}

.quarto-listing {
  margin-top: 2em;
}

.quarto-listing .listing-name,
.quarto-listing .listing-author {
  white-space: nowrap;
}

.quarto-listing .listing-actions-group h3 {
  margin-top: 0;
}

 
</style>
<ul id="index-chooser" class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation">
    <a class="nav-link" href="data-asv.html">
      <i class="bi bi-database-fill"></i>16S rRNA ASV processing
    </a>
  </li>
  <li class="nav-item" role="presentation">
    <a class="nav-link" href="data-otu.html">
      <i class="bi bi-database-fill"></i>16S rRNA OTU processing
    </a>
  </li>
  <li class="nav-item" role="presentation">
    <a class="nav-link" href="data-med.html">
      <i class="bi bi-database-fill"></i>16S rRNA MED processing
    </a>
  </li>
  <li class="nav-item" role="presentation">
    <a class="nav-link" href="data-mg.html">
      <i class="bi bi-database-fill"></i>Metagenomic processing
    </a>
  </li>
</ul>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  // get file name
  const filename = window.location.pathname.split("/").slice(-1)[0];

  // latch active
  const toolLinks = window.document.querySelectorAll("#index-chooser a");
  for (const tool of toolLinks) {
    if (filename && filename !== "index.html") {
      if (tool.href.endsWith(filename)) {
        tool.classList.add("active");
      } 
    } else {
      if (tool.href.endsWith("listing-filters.html")) {
        tool.classList.add("active");
      }
    }
  }
  
  // move heading into table
  document.querySelector(".listing-actions-group").prepend(document.querySelector("h3.unlisted"));
});

</script>
</div>
<p><br></p>
<section id="med-data-and-scripts" class="level2 unlisted page-columns page-full"><h2 class="unlisted" data-anchor-id="med-data-and-scripts">MED Data and Scripts</h2>
<div id="listing-med-listing" class="column-body-outset-right quarto-listing quarto-listing-container-table">
<table class="quarto-listing-table table">
<thead><tr>
<th>
Name
</th>
<th>
Description
</th>
</tr></thead>
<tbody class="list">
<tr data-index="0" data-listing-name-sort="Sequence data">
<td>
<a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB36632" class="name listing-name">Sequence data</a>
</td>
<td>
<span class="listing-description">Output files from alignment step of the mothur OTU pipeline for 1809 samples </span>
</td>
</tr>
<tr data-index="1" data-listing-name-sort="Sample data">
<td>
<a href="include/all_metadata.txt.zip" class="name listing-name">Sample data</a>
</td>
<td>
<span class="listing-description">16S rRNA sample data </span>
</td>
</tr>
<tr data-index="2" data-listing-name-sort="mothur processing scripts">
<td>
<a href="include/scripts/med/med_batchfile_processing.zip" class="name listing-name">mothur processing scripts</a>
</td>
<td>
<span class="listing-description">The mothur portion of the MED pipeline </span>
</td>
</tr>
<tr data-index="3" data-listing-name-sort="Data prep scripts">
<td>
<a href="include/scripts/med/mothur2oligo.zip" class="name listing-name">Data prep scripts</a>
</td>
<td>
<span class="listing-description">The mothur2oligo scripts needed to prep the data for MED analysis </span>
</td>
</tr>
<tr data-index="4" data-listing-name-sort="Mapping file">
<td>
<a href="include/scripts/med/med_mapping.txt.zip" class="name listing-name">Mapping file</a>
</td>
<td>
<span class="listing-description">The sample mapping needed to run the MED analysis </span>
</td>
</tr>
<tr data-index="5" data-listing-name-sort="Hydra job scripts">
<td>
<a href="include/scripts/med/med_hydra_scripts.zip" class="name listing-name">Hydra job scripts</a>
</td>
<td>
<span class="listing-description">Hydra job scripts. </span>
</td>
</tr>
<tr data-index="6" data-listing-name-sort="Pipeline read changes">
<td>
<a href="include/scripts/med/all_sample_med_read_changes.txt.zip" class="name listing-name">Pipeline read changes</a>
</td>
<td>
<span class="listing-description">Table showing read changes for each sample through the processing pipeline </span>
</td>
</tr>
</tbody>
</table>
<div class="listing-no-matching d-none">
No matching items
</div>
</div>
</section><section id="med-processing" class="level2 page-columns page-full"><h2 data-anchor-id="med-processing">MED Processing</h2>
<p>Processing scripts for <a href="https://merenlab.org/2014/11/04/med/">Minimum Entropy Decomposition (MED)</a> analysis. The pipeline begins with the output fasta and count files from the <code>align.seqs</code> part of the mothur OTU pipeline. From there we use mothur to remove negative control samples, check for chimera, and run taxonomic classifications. It is important to note that the MED workflow does not precluster sequences (as in the mothur pipeline) because MED relies on every sequence (including redundant reads) for the analysis. This pipeline has four main steps:</p>
<ol type="1">
<li>run the mothur workflow.<br>
</li>
<li>modify and run the <code>mothur2oligo.sh</code> script. This script transforms the mothur output to appropriate format for MED. It must be run in the mothur environment because the script uses mothur. We need access to the following mothur files to run this script.
<ol type="i">
<li>
<strong>taxonomy file</strong>: <code>final_med.taxonomy</code><br>
</li>
<li>
<strong>count file</strong>: <code>final_med.count_table</code><br>
</li>
<li>
<strong>fasta file</strong>: <code>final_med.fasta</code><br>
</li>
</ol>
</li>
<li>trim uninformative columns from alignment (in the MED environment)<br>
</li>
<li>run the MED command</li>
</ol>
<div class="cell">
<details class="code-fold"><summary>mothur processing script for MED analysis</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">set.dir</span><span class="er">(</span><span class="va">output</span><span class="op">=</span>/pool/genomics/stri_istmobiome/data/TRANS_SHRIMP/16S/MOTHUR/pipelineFiles_med/<span class="kw">)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="ex">system</span><span class="er">(</span><span class="fu">cp</span> pipelineFiles/PROCESSING_FILES/shrimp.trim.contigs.good.unique.good.filter.unique.fasta pipelineFiles_med/<span class="kw">)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="ex">system</span><span class="er">(</span><span class="fu">cp</span> pipelineFiles/PROCESSING_FILES/shrimp.trim.contigs.good.unique.good.filter.count_table pipelineFiles_med/<span class="kw">)</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">############################################################</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">### Just copied the fasta and count files from last command</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">############################################################</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">############################################################</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">### ############  FOR MED REMOVED PRECLUST #################</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">############################################################</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">############################################################</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">### ############  REMOVE NEGATIVE CONTROL ##################</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">### from https://forum.mothur.org/t/negative-control/2754/16</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">############################################################</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="ex">get.groups</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.count_table, <span class="va">groups</span><span class="op">=</span>Control_49-Control_23-Control_17-Control_24-Control_14-Control_44-Control_20-Control_33-Control_41-Control_29-Control_50-Control_22-Control_19-Control_18-Control_48-Control_13-Control_21-Control_16-Control_30-Control_5-Control_42-Control_25-Control_51-Control_40-Control_15-Control_36-Control_47-Control_27-Control_32-Control_8-Control_3-Control_4-Control_6-Control_45-Control_26-Control_46-Control_53-Control_7-Control_12-Control_10-Control_9-Control_35-Control_54-Control_2-Control_43-Control_1-Control_11-Control_52-Control_38-Control_34-Control_56-Control_37-Control_28-Control_57-Control_31-Control_39-Control_59-Control_55-Control_60-Control_58<span class="kw">)</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">input</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, <span class="va">new</span><span class="op">=</span>neg_control.fasta<span class="kw">)</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">input</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.count_table, <span class="va">new</span><span class="op">=</span>neg_control.count_table<span class="kw">)</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>neg_control.fasta, <span class="va">count</span><span class="op">=</span>neg_control.count_table, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="ex">list.seqs</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>neg_control.count_table<span class="kw">)</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################################################</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co">### modified neg_control.count_table: </span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co">### 1. remove first two rows</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co">### 2. in BBEdit run \t\d+,.*</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co">### 3. in R</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co">###     library(tidyverse)</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="co">###     a &lt;- read_tsv("neg_control.count_table")</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co">###     b &lt;- read_tsv("shrimp.trim.contigs.good.unique.good.filter.count_table")</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="co">###     e &lt;- dplyr::left_join(a, b, by = "Representative_Sequence")</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co">###     write.table(e, "results.txt", row.names = FALSE, quote = FALSE, sep = "\t")</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co">### 4. Replace accnos file with new list. Removed sequences that were found less than 10% in Neg control</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################################################</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.seqs</span><span class="er">(</span><span class="va">accnos</span><span class="op">=</span>neg_control_subset.accnos, <span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.count_table<span class="kw">)</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co">## ADDED this command to remove NC samples</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co">## [ERROR]: Control_18 is not in your count table. Please correct.</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="co">## [ERROR]: Control_5 is not in your count table. Please correct.</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.count_table, <span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, <span class="va">groups</span><span class="op">=</span>Control_1-Control_10-Control_11-Control_12-Control_13-Control_14-Control_15-Control_16-Control_17-Control_19-Control_2-Control_20-Control_21-Control_22-Control_23-Control_24-Control_25-Control_26-Control_27-Control_28-Control_29-Control_3-Control_30-Control_31-Control_32-Control_33-Control_34-Control_35-Control_36-Control_37-Control_38-Control_39-Control_4-Control_40-Control_41-Control_42-Control_43-Control_44-Control_45-Control_46-Control_47-Control_48-Control_49-Control_50-Control_51-Control_52-Control_53-Control_54-Control_55-Control_56-Control_57-Control_58-Control_59-Control_6-Control_60-Control_7-Control_8-Control_9<span class="kw">)</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="co">### NEGATIVE CONTROLS Should be GONE </span><span class="al">###</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="ex">chimera.vsearch</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table, <span class="va">dereplicate</span><span class="op">=</span>t, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="ex">classify.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, <span class="va">reference</span><span class="op">=</span>reference_dbs/gsrdb.fasta, <span class="va">taxonomy</span><span class="op">=</span>reference_dbs/gsrdb.tax, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.lineage</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, <span class="va">taxonomy</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.taxonomy, <span class="va">taxon</span><span class="op">=</span>Chloroplast-Mitochondria-unknown-Eukaryota<span class="kw">)</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.tax</span><span class="er">(</span><span class="va">taxonomy</span><span class="op">=</span>current, <span class="va">count</span><span class="op">=</span>current, <span class="va">processors</span><span class="op">=</span>30<span class="kw">)</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>current<span class="kw">)</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="co">##########################</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.pick.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table, <span class="va">taxonomy</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.pick.taxonomy, <span class="va">prefix</span><span class="op">=</span>final_med<span class="kw">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Once you have the script and data you simply run the pipeline like so.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mothur</span> med_batchfile_processing</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Once the mothur portion of the workflow is complete, the script <a href="https://github.com/DenefLab/MicrobeMiseq/tree/master/mothur2oligo"><code>mothur2oligo.sh</code></a> needs to be run in the <code>mothur</code> environment and modified for your specific purposes. You should not need to modify the associated <code>renamer.pl</code> script but it does need to be in the same location as <code>mothur2oligo.sh</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> mothur2oligo.sh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">mothur2oligo</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">renamer</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<details class="code-fold"><summary>Expand for the <code>mothur2oligo.sh</code> script</summary><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate mothur</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># USAGE: sh mothur2oligo.sh</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is a shell script for transforming mothur output to appropriate format for </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># A. murat Eren's oligotyping pipeline </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">## Set variables</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust the file names to your own study - these are the files from the mothur SOP</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="va">taxonomy</span><span class="op">=</span><span class="st">"pipelineFiles_med/final_med.taxonomy"</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="va">fasta</span><span class="op">=</span><span class="st">"pipelineFiles_med/final_med.fasta"</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="va">count</span><span class="op">=</span><span class="st">"pipelineFiles_med/final_med.count_table"</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="va">processors</span><span class="op">=</span>28</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the taxon you want to select, separate taxonomic levels with ";" </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Do not touch inner and outer quotes</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="va">taxon</span><span class="op">=</span><span class="st">"'Bacteria;-Archaea;'"</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">########## Script  #############</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="va">redundantFasta</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta}</span>.pick.redundant.fasta <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.fasta//'</span><span class="va">)</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="va">groups</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${count}</span>.pick.redundant.groups <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.count_table//'</span><span class="va">)</span> </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Call mothur commands for generating deuniqued fasta file for a specific lineage</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="ex">mothur</span> <span class="st">"#set.current(processors=</span><span class="va">$processors</span><span class="st">); get.lineage(taxonomy=</span><span class="va">$taxonomy</span><span class="st">, taxon=</span><span class="va">$taxon</span><span class="st">, count=</span><span class="va">$count</span><span class="st">); list.seqs(count=current); get.seqs(accnos=current, fasta=</span><span class="va">$fasta</span><span class="st">); deunique.seqs(fasta=current, count=current)"</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace all "_" in fasta header with a ":"</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">$groups</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/_/:/g'</span> <span class="op">&gt;</span> intermediate1</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a file which maps sample names to sequence headers</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span> <span class="va">$groups</span> intermediate1 <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'BEGIN{FS="\t"}{print $1"\t"$2"_"$3}'</span> <span class="op">&gt;</span> intermediate2</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Perl script to rename the headers of the fasta to include the sample name at the beginning followed by a "_"</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="fu">perl</span> renamer.pl <span class="va">$redundantFasta</span> intermediate2 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<details class="code-fold"><summary>and the companion <code>renamer.pl</code> script</summary><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate mothur</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#! /usr/bin/perl</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#from http://www.perlmonks.org/?node_id=975419</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="ex">use</span> strict<span class="kw">;</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="ex">use</span> warnings<span class="kw">;</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="ex">@ARGV</span> == 2 or die <span class="st">"usage: </span><span class="va">$0</span><span class="st"> &lt;multifasta file&gt; &lt;header replacement fil</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="st">+e&gt;\n"</span><span class="kw">;</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="ex">my</span> <span class="er">(</span> <span class="va">$fasta_file</span><span class="ex">,</span> <span class="va">$header_file</span> <span class="kw">)</span> <span class="ex">=</span> @ARGV<span class="kw">;</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="ex">my</span> <span class="va">$destination</span> = <span class="va">$fasta_file</span> . <span class="st">'_headers-replaced.fasta'</span><span class="kw">;</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> IN2, <span class="st">'&lt;'</span>, <span class="va">$header_file</span> or die <span class="st">"Can't read from tab-delimited head</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="st">+er replacement file </span><span class="va">$header_file</span><span class="st">: </span><span class="va">$!</span><span class="st">\n"</span><span class="kw">;</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="ex">my</span> %head_seqs<span class="kw">;</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="kw">(</span> <span class="op">&lt;</span>IN2<span class="op">&gt;</span> <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    <span class="ex">chomp</span><span class="kw">;</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="ex">my</span> <span class="er">(</span> <span class="va">$old</span><span class="ex">,</span> <span class="va">$new</span> <span class="kw">)</span> <span class="ex">=</span> split /\t/<span class="kw">;</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    <span class="va">$head_seqs</span><span class="ex">{</span> <span class="va">$old</span> } = <span class="va">$new</span><span class="kw">;</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="kw">}</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="ex">close</span> IN2<span class="kw">;</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> IN1, <span class="st">'&lt;'</span>, <span class="va">$fasta_file</span> or die <span class="st">"Can't read from multifasta file wit</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="st">+h alternating lines of headers and sequences </span><span class="va">$fasta_file</span><span class="st">: </span><span class="va">$!</span><span class="st">\n"</span><span class="kw">;</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> OUT, <span class="st">'&gt;'</span>, <span class="va">$destination</span> or die <span class="st">"Can't write to file </span><span class="va">$destination</span><span class="st">: </span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="st">+</span><span class="va">$!</span><span class="st">\n"</span><span class="kw">;</span>    </span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="kw">(</span> <span class="op">&lt;</span>IN1<span class="op">&gt;</span> <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">(</span> <span class="ex">/^</span><span class="op">&gt;(</span><span class="bu">.</span>+<span class="op">)</span>$/ <span class="kw">&amp;&amp;</span> <span class="ex">exists</span> <span class="va">$head_seqs</span>{ <span class="va">$1</span> } <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>        <span class="va">$_</span> = <span class="st">"&gt;</span><span class="va">$head_seqs</span><span class="st">{ </span><span class="va">$1</span><span class="st"> }\n"</span><span class="kw">;</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>        <span class="kw">}</span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>    <span class="ex">print</span> OUT <span class="va">$_</span><span class="kw">;</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    <span class="kw">}</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="ex">close</span> IN1<span class="kw">;</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="ex">close</span> OUT<span class="kw">;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
</div>
</div>
<p>Great. Now within the oligotype/MED environment run the following commands for the MED analysis. You will need the <code>mapping.txt</code> file linked above for this step.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate oligotyping</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">o-trim-uninformative-columns-from-alignment</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>        final_med.pick.redundant.fasta_headers-replaced.fasta</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="ex">decompose</span> final_med.pick.redundant.fasta_headers-replaced.fasta-TRIMMED <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">-E</span> mapping.txt <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output-directory</span> MED <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">--number-of-threads</span> 24 <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">--skip-gen-figures</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the resources listed above, we include a table that summarizes read changes for each sample through the pipeline.</p>
<div class="column-body-outset-right">

</div>



</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/istmobiome\.github\.io\/trans-shrimp\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Â© Copyright 2024 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; <a href="https://istmobiome.rbind.io/">The Istmobiome Project</a>. Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
<div class="toc-actions"><ul><li><a href="https://github.com/istmobiome/trans-shrimp/edit/main/docs/data/data-med.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/blob/main/docs/data/data-med.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>


</body></html>